astroid/nodes/as_string.py

```diff
@@ -50,15 +51,17 @@ def _docs_dedent(self, doc_node: Const | None) -> str:
             self.indent, doc_node.value.replace("\n", DOC_NEWLINE)
         )
 
-    def _stmt_list(self, stmts, indent=True):
+    def _stmt_list(self, stmts: list, indent: bool = True) -> str:
         """return a list of nodes to string"""
-        stmts = "\n".join(nstr for nstr in [n.accept(self) for n in stmts] if nstr)
-        if indent:
-            return self.indent + stmts.replace("\n", "\n" + self.indent)
+        stmts_str: str = "\n".join(
```

Comment:
Mypy doesn't like variables changed to different types.

Reply:
Let's use two variable name instead. This is a pattern that is used a lot in old python but it's very error prone, imo it dying with mypy's adoption is a good thing).